src/components/textfieldButtons/js/TextfieldButtons.js

```diff
@@ -0,0 +1,166 @@
+            },
+            // TODO: This listener is to deal with the issue that, when the input field receives a invalid input such as a string value,
+            // ignore it and populate the field with the previous value.
+            // This is an area in which UX has spilled over into our model configuration, which to some extent we should try to prevent.
+            // Whenever we receive a "change" event or some other similar checkpoint, if these updates occurred any faster, the user would
+            // be infuriated by being unable to type into the field. This situation doesn't occur at the moment because the change event is
+            // only fired when users leave the input feild. At the very least, we need to give a namespace to this listener - unfortunately
+            // the current dataBinding implementation will ignore it. Having this listener here represents an interaction decision rather
+            // than an implementation decision. This issue needs to be revisited.
+            stringValue: {
+                "this": "{that}.container",
+                "method": "val",
+                args: ["{that}.model.value"]
+            }
+        },
+        listeners: {
+            "onCreate.bindChangeEvt": {
+                "this": "{that}.container",
+                "method": "change",
+                "args": ["{that}.setModel"]
+            }
+        },
+        invokers: {
+            setModel: {
+                changePath: "stringValue",
+                value: "{arguments}.0.target.value"
+            }
+        }
+    });
+
+    fluid.defaults("fluid.stepper.button", {
+        gradeNames: ["fluid.viewComponent"],
+        range: {}, // should be used to specify the min, max range e.g. {min: 0, max: 100}
+        invokers: {
+            setModel: {
+                changePath: "value",
+                value: {
+                    expander: {
+                        funcName: "{that}.calculateValue"
+                    }
+                }
+            },
+            calculateValue: {
+                funcName: "fluid.stepper.button.buttonCalculateValue",
+                args: ["{that}.options.incrementCoefficient", "{that}.options.buttonOptions.stepMultiplier", "{that}.model.value", "{that}.options.range.min", "{that}.options.range.max"]
+            }
+        },
+        listeners: {
+            "onCreate.bindClickEvt": {
+                "this": "{that}.container",
+                "method": "click",
+                "args": "{that}.setModel"
+            }
+        },
+        incrementCoefficient: 1,
+        buttonOptions: {
+            stepMultiplier: 10
+        }
+    });
+
+    fluid.stepper.button.buttonCalculateValue = function (value, stepMultiplier, modelValue, min, max) {
+        modelValue = Math.round(modelValue * stepMultiplier);
+        min = Math.round(min * stepMultiplier);
+        max = Math.round(max * stepMultiplier);
+        if ((modelValue + value) < min) {
```

Comment:
I am not entirely sure how the limit Range works. Does it return the max value if the supplied value is greater than the max and what is its syntax? I use stepMultipliers to avoid rounding errors. We discussed that with jobara here.